spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessor.java

```diff
@@ -236,21 +239,55 @@ public void doWith(Method method) throws IllegalArgumentException, IllegalAccess
 				}
 			}
 		});
-		if (classLevelListener != null) {
-			processMultiMethodListener(classLevelListener, multiMethods, bean, beanName);
+		if (hasClassLevelListeners) {
+			processMultiMethodListeners(classLevelListeners, multiMethods, bean, beanName);
 		}
 		return bean;
 	}
 
-	private void processMultiMethodListener(RabbitListener classLevelListener, List<Method> multiMethods, Object bean,
-			String beanName) {
+	/*
+	 * AnnotationUtils.getRepeatableAnnotations does not look at interfaces
```

Comment:
If you guys need an `AnnotationUtils.findRepeatableAnnotations(...)` variant (note "find" vs "get"), please just open a JIRA issue against Spring Framework, and feel free to assign it to me.